src/main/webapp/js/user-page-loader.js

```diff
@@ -107,10 +107,17 @@ function fetchAboutMe(){
   })
 }
 
+function addRichTextEditor(){
+  const config = {removePlugins: [ 'ImageUpload']};
+  ClassicEditor.create( document.getElementById('message-input', config) );
+  ClassicEditor.create( document.getElementById('about-me-input', config) );
```

Comment:
This will be a learning moment for me haha, I'm a bit confused on where the document variable comes into play. Would you mind telling me what/where what exactly is it, and what does document.getElementById('message-input', config) return? Also, just making sure, is config a key-value pair passed into the above function?

Reply:
Honestly, I'm not super sure about the document variable; I was mostly following instructions here. However, it looks like document is already being used throughout the file, in things like setPageTitle(), showFormsIfViewingSelf(), and fetchMessages(). It looks like "document" comes from the HTML, [This page](https://www.w3schools.com/js/js_htmldom_document.asp) says "The HTML DOM document object is the owner of all other objects in your web page." so I guess it's the page as a whole? 

Actually, you just caught my error, thank you. That config is supposed to go outside of the getElementByID. The getElement... returns the "about me" element, but we need to configure the editor.

As for config, I'm not familiar with JS, but I think so. It looks like the JS dictionary format that I'm seeing in a google search, and that exact syntax looks identical to python. That being said, the "value" is an array, I think. So it's a list of plugins to remove, with the key "removePlugins." I think,

Reply:
@nbaldy has described this pretty well. I'll just add a few extra details.

document is basically a global variable. It is like a public static variable in Java (e.g. System.out) or a variable in the global namespace in C++ (e.g. std::cout, std::cerr). It basically can be accessed in JavaScript code from almost anywhere and it represents the current document. You can use it to access all aspects of the HTML document. document.getElementById() lets you search the HTML on the page for an element with  a specific ID. 

The config is technically a Javascript object, but I agree it acts much like a python dictionary. Just like python, you can access it like an object (e.g. config.removePlugins[0]) or like a dictionary (e.g.  config["removePlugins"][0]) There is some subtlety that I admit I don't fully understand, but this is a good mental model to start with. I tend to like using object syntax because it is less characters and usually I don't have whitespace or other separators like "-" or "+" in the keys.

Hopefully this helps.